main/src/ca/uwaterloo/flix/api/lsp/provider/CompleteProvider.scala

```diff
@@ -54,8 +116,34 @@ object CompleteProvider {
     // Declaration-based:
 
     // Expressed-based:
-    CompletionItem("match", "match ${1:exp} {\n case ${2:pat} => ${3:exp}\n}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
-    CompletionItem("query", "query ${1:db} select ${2:cols} from ${3:preds} ${4:where ${5:cond}}", None, None, CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("def", "def ${1:name}(${2:arg}:${3:type}): ${4:return_type} = \n", None, Some("code snippet to define a function"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("elif", "else if (${1:/* condition */}) {\n    ${2:/* code */}\n}", None, Some("code snippet for else if ()"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("else", "else {\n    ${1:/* code */}\n}", None, Some("code snippet for else"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("enum", "enum ${1:name} {\n    case ${2:DataType1}\n    ${4:/* add more algebric data types separated by comma (,) */}\n}", None, Some("code snippet for enum"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("if", "if (${1:/* condition */}) {\n    ${2:/* code */}\n}", None, Some("code snippet for if ()"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import ${1:method}", None, Some("import code snippet for method call"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import new ${1:object} as ${2:name}", None, Some("import code snippet for new object"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import get ${1:field} as ${2:name}", None, Some("import code snippet to get field"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("import", "import set ${1:field} as ${2:name}", None, Some("import code snippet to set field"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
+    CompletionItem("instance", "instance ${1:name}[Option[${2:a}]] with $1[$2] {\n    ${3:/* code */}\n}", None, Some("code snippet for instance"), CompletionItemKind.Snippet, InsertTextFormat.Snippet, Nil),
```

Comment:
The form here should be more like:

instance ${1:typeclass}[${2:type}] 

then the with part should be optional with nested use of ${}.